File: platform/Codewars/src/vowel.py
def encode(string):
    vowel_map = {'a':1, 'e':2, 'i':3, 'o':4, 'u':5}
    return ''.join([str(vowel_map[letter]) if letter in vowel_map else letter for letter in string])

def decode(string):    
    vowel_map = {'1':'a', '2':'e', '3':'i', '4':'o', '5':'u'}
    return ''.join([str(vowel_map[letter]) if letter in vowel_map else letter for letter in string])
    
if __name__ == "__main__":
    selection = input("Enter 1 for encoding, or 2 for decoding: ")
    while (int(selection) not in [1,2]):
        print('Your selection is invalid please choose again.')
        selection = input("Enter 1 for encoding, or 2 for decoding: ")
        
    words_to_code = input("Enter your phrase: ")
    print(encode(words_to_code)) if selection == '1' else print(decode(words_to_code))
    
    

File: platform/Codewars/src/Vowel.java
import java.util.HashMap;
import java.util.Map;
import java.util.Scanner;

public class Vowel {
    
    public static void main(String[] args) {
        System.out.println("Please enter a phrase/string:");
        Scanner input = new Scanner(System.in);
        String phrase = input.nextLine();
        
        System.out.println("The string can be encoded as: " + encode(phrase));
        System.out.println("But can be decoded back into: " + decode(encode(phrase)));
            
    }
    
    public static String encode(String words) {
        Map<Character, Integer> vowels = new HashMap<>();
        vowels.put('a', 1);
        vowels.put('e', 2);
        vowels.put('i', 3);
        vowels.put('o', 4);
        vowels.put('u', 5);

        StringBuilder output = new StringBuilder();
        for (char letter: words.toCharArray()) {
            if (vowels.containsKey(letter)) {
                output.append(vowels.get(letter));
            } else {
                output.append(letter);
            }
        }
        return output.toString();
    }
    
    public static String decode(String words) {
        Map<Character, Character> vowels = new HashMap<>();
        vowels.put('1', 'a');
        vowels.put('2', 'e');
        vowels.put('3', 'i');
        vowels.put('4', 'o');
        vowels.put('5', 'u');

        StringBuilder output = new StringBuilder();
        for (char letter: words.toCharArray()) {
            if (vowels.containsKey(letter)) {
                output.append(vowels.get(letter));
            } else {
                output.append(letter);
            }
        }
        return output.toString();
    }
}